### DISCUSSION TITLE
NaiveProxy 服务端插件使用官方指定源码还是 SagerNet 大源码？

### DISCUSSION THREAD

**Discussion by lxhao61** [lxhao61/integrated-examples#201] [Polls]
目前 NaiveProxy 服务端的 forwardproxy 插件可以使用官方指定（klzgrad 大）源码与 SagerNet 大源码。现将官方指定（klzgrad 大）源码与 SagerNet 大源码优劣告知，请大家参与投票。

1、使用官方指定（klzgrad 大）源码更新及时，但暂不支持 UoT。
--with github.com/caddyserver/forwardproxy=github.com/klzgrad/forwardproxy@naive

2、使用 SagerNet 大源码支持 UoT，但长时间没有更新了。
--with github.com/caddyserver/forwardproxy=github.com/sagernet/forwardproxy@latest

**Comment by Bryan2333** [comment]
我个人觉得还是使用官方指定的分支比较好，SagerNet的分支已经两年没更新了。另外，klzgrad的分支添加了[Naive padding protocol v1](https://github.com/klzgrad/forwardproxy/commit/e8833020ffbcbe45d210b03255364f48947b3175)的支持

**Comment by pureair** [comment]
看了这个才知道2.7.6开启UoT客户端可以使用SagerNet。之前用的Nekobox里的UoT (标注了sing-box server)，实测2.7.6不兼容，还以为没有这个功能。不过我研究了一下SagerNet和Nekobox的navie-proxy用的是同一个插件(sagernet的[NaïveProxy Plugin](https://github.com/SagerNet/SagerNet/releases?q=naive), 也不知道为什么一个能代理udp另一个不行。
